arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/precompiles/Sha256Blocks.java

```diff
@@ -45,6 +43,17 @@ public String moduleKey() {
   private static final int SHA256_PADDING_LENGTH = 64;
   private static final int SHA256_NB_PADDED_ONE = 1;
 
+  private final Hub hub;
+  private final Stack<Integer> counts = new Stack<>();
+
+  @Getter private final ShakiraData data;
+  private int lastDataCallHubStamp = 0;
```

Comment:
Same, it's strange that it's stored here

Reply:
There really should only be a single kind of SHAKIRA operation with some field with values in an `enum` that specifies the underlying operation i.e. either `KECCAK`, `SHA2-256` or `RIPEMD-160`. Everything is derivable from that.